### PR TITLE
Postgres impl

### PR DESCRIPTION
**Pull request by jarri-abidi**

### CODE REVIEW

File: checklist/endpoint.go [hunk 1]
```diff
@@ -10,96 +10,94 @@ import (
...
+}
+
+func (r saveTaskResponse) Failed() error { return r.Err }
+
+type listTasksResponse struct {
+	Tasks []taskResponse
+	Err   error `json:"error,omitempty"`
+}
+
+func (r listTasksResponse) MarshalJSON() ([]byte, error) {
```
Comment (markhaur) [inline checklist/endpoint.go:82]: that's interesting. defining our own MarshalJSON on our response dto.

File: checklist/endpoint.go [hunk 2]
```diff
@@ -10,96 +10,94 @@ import (
...
-
 func makeListTasksEndpoint(s Service) endpoint.Endpoint {
 	ep := func(ctx context.Context, _ interface{}) (interface{}, error) {
 		tasks, err := s.List(ctx)
-		var resp = listTasksResponse{Tasks: make([]struct {
-			ID   int64  `json:"id"`
-			Name string `json:"name"`
-			Done bool   `json:"done"`
-		}, 0), Err: err}
+		var resp = listTasksResponse{Tasks: make([]taskResponse, 0), Err: err}
```
Comment (markhaur) [inline checklist/endpoint.go]:
```suggestion
		var resp = listTasksResponse{Tasks: make([]taskResponse, 0, len(tasks)), Err: err}
```

File: checklist/endpoint.go (hunk 1 above)
Comment (jarri-abidi) [reply checklist/endpoint.go:82]: yep.. this gets automatically used by encoding/json pkg whenever we try to decode this struct

**Review by markhaur — APPROVED** [review]
looks good to me. But we have to work on inmem database as a default choice to run our app.

File: Makefile [hunk 3]
```diff
@@ -2,11 +2,18 @@
...
 
 test:
 	go test -cover ./...
 
 run: build
 	./app
+
+clean:
+	rm app
+	rm -r postgres/gen/**
```
Comment (jarri-abidi) [inline Makefile]:
```suggestion
	rm app && rm -r postgres/gen/**
```

**Pull request merged**